File: scripts/test_solutions/1584_min_cost_connect_points.py
def minCostConnectPoints(points):
    n = len(points)
    visited = set()
    min_dist = [float('inf')] * n
    min_dist[0] = 0
    cost = 0

    for _ in range(n):
        u = -1
        min_val = float('inf')
        for j in range(n):
            if j not in visited and min_dist[j] < min_val:
                min_val = min_dist[j]
                u = j
        visited.add(u)
        cost += min_val
        for v in range(n):
            if v not in visited:
                dist = abs(points[u][0] - points[v][0]) + abs(points[u][1] - points[v][1])
                min_dist[v] = min(min_dist[v], dist)

    return cost

File: scripts/test_solutions/1584_min_cost_connect_points.js
var minCostConnectPoints = function(points) {
    const n = points.length, visited = new Set(), minDist = Array(n).fill(Infinity);
    minDist[0] = 0;
    let cost = 0;
    for (let i = 0; i < n; i++) {
        let u = -1, minVal = Infinity;
        for (let j = 0; j < n; j++) if (!visited.has(j) && minDist[j] < minVal) { minVal = minDist[j]; u = j; }
        visited.add(u);
        cost += minVal;
        for (let v = 0; v < n; v++) {
            if (!visited.has(v)) {
                const dist = Math.abs(points[u][0] - points[v][0]) + Math.abs(points[u][1] - points[v][1]);
                minDist[v] = Math.min(minDist[v], dist);
            }
        }
    }
    return cost;
};
